R/testing.R

```diff
@@ -43,7 +43,8 @@ run_test <- function(hook_name,
                      artifacts = NULL,
                      file_transformer = function(files) files,
                      env = character(),
-                     expect_success = is.null(std_err)) {
+                     expect_success = is.null(std_err),
+                     post_hook_assert = NULL) {
```

Comment:
I think you can default to `invisible` (the function) and avoid the if statement further down.

Reply:
actually no wait a sec.

Reply:
That sounds reasonable. To paraphrase: we add an additional assertion to `hook_state_assert_one` based on `read_only` argument that is derived from the nature of the hook? In this case, of testing spell-check, we would test it in both scenarios. Is that right? It might make sense to rename the flag to `--read-only`, so it is more generic and can be shared across other hooks.

I have started working on the change as I understood it and outlined above, however I keep running into roxygenize tests failing, even on a clean main branch.
```
── Failed tests ── Failed tests ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-hooks.R:394:1): (code run outside of `test_that()`)
<purrr_error_indexed/rlang_error/error/condition>
Error in `purrr::map2(path_candidate, path_candidate_temp, hook_state_assert_one,
    tempdir = tempdir, file_transformer = file_transformer, path_stdout = path_stdout,
    path_stderr = path_stderr, expect_success = expect_success,
    std_err = std_err, std_out = std_out, exit_status = exit_status)`: ℹ In index: 1.
ℹ With name: man/flie.Rd.
Caused by error:
! in/flie.Rd and /private/var/folders/lh/y7zmk_td7nzfjxxqxdwhy5r40000gn/T/RtmpLxX4J1/fileeaf15c5ae334/man/flie.Rd are not supposed to be identical but they are

[ FAIL 1 | WARN 0 | SKIP 1 | PASS 71 ]
```
Do you have an idea where that comes from?

Reply:
All great, exactly. This failure occurred as well in previous CI runs. Don’t know why. Try updating roxygen and other deps. Otherwise You can ignore it.

Reply:
Correct! I have a rough draft for the testing framework change, but haven't had an opportunity to clean it up and publish just yet.